presto-main/src/main/java/com/facebook/presto/operator/MergeHashSort.java

```diff
@@ -45,11 +45,23 @@ public MergeHashSort(AggregatedMemoryContext memoryContext)
     /**
      * Rows with same hash value are guaranteed to be in the same result page.
      */
-    public WorkProcessor<Page> merge(List<Type> keyTypes, List<Type> allTypes, List<WorkProcessor<Page>> channels, DriverYieldSignal driverYieldSignal)
+    public WorkProcessor<Page> merge(List<Type> keyTypes, List<Type> allTypes, List<WorkProcessor<Page>> pages, DriverYieldSignal driverYieldSignal)
     {
-        InterpretedHashGenerator hashGenerator = InterpretedHashGenerator.createPositionalWithTypes(keyTypes);
+        return merge(keyTypes, null, allTypes, pages, driverYieldSignal);
+    }
+
+    public WorkProcessor<Page> merge(List<Type> keyTypes, List<Integer> keyChannels, List<Type> allTypes, List<WorkProcessor<Page>> pages, DriverYieldSignal driverYieldSignal)
+    {
+        InterpretedHashGenerator hashGenerator;
+        if (keyChannels == null) {
```

Comment:
we do not pass nulls around; it's dangerous. But coming back to the logic, could you explain why we need `keyChannels`? I haven't read the callsites of it. But I would assume the merge should be similar to aggregation so curious why we need `keyChannels`

Reply:
Good point. So IIUC, InterpretedHashGenerator has 2 ways of instantiation 
1. Specify only `hashTypes` - in which case it assumes that the first N columns/channels are assumed to be the `hashChannels`. HashAggregations fit this criteria and hence SpillableHashAggregationBuilder instantiates it through this approach

2. Specify `hashTypes`, `hashChannels` - This is used by TopN, because the first N columns are not necessarily the grouping keys, so we specify explicitly

Reply:
About passing nulls, I see that the underlying InterpretedHashGenerator class has multiple constructors, following kind of similar approach of having nullable parameters. This change is a side effect of that. I have added a `@Nullable` annotation to the parameter to make it safer. WDYT ?

Reply:
I think it might be fine given `InterpretedHashGenerator` has nullables. But can we have explicitly comments on the method?

Reply:
Makes sense. I have added comments on the branch (if) condition to keep it simple. Also added `@Nullable` annotation to the parameter.